webapp/src/components/Vault/VaultActionsForm/Modal/ActionSteps.tsx

```diff
@@ -329,9 +330,16 @@ const ActionSteps: React.FC<ActionStepsProps> = ({
                     switch (vaultActionForm.vaultOption) {
                       case "rSOL-THETA":
                         if (client) {
+                          const pricePerShare = await getSOLPricePerShare();
+                          const amountToRedeem = Math.floor(
+                            (Number(vaultActionForm.inputAmount) /
+                              Number(pricePerShare)) *
+                              Math.pow(10, decimals)
+                          );
```

Comment:
It makes sense to just abstract this calculation out, similar to [ShareMath assetToShares](https://github.com/ribbon-finance/ribbon-v2/blob/master/contracts/libraries/ShareMath.sol#L12)